### PR TITLE
docs: Add Tabs documentation

### PR DESCRIPTION
**Pull request by didoesdigital**
Branch preview: https://dev.cultureamp.design/dst/add-tab-documentation/components/tabs

### CODE REVIEW

File: site/docs/components/tabs.mdx [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+    - Selected tab: The current, active tab. One tab—and only one tab—is always selected for a given tab list.
+    - Unselected tab: The other available tabs.
+    - Tab item:
+        - A tab item contains a large clickable target area, a label, a selected tab indicator (when the tab is selected), and optional metadata.
+        - Tab link: The large clickable area of the tab item uses a link to navigate to that tab's content.
+        - Tab label: The label describes what content will be displayed. Use 1 short noun or noun phrase.
+        - Selected tab indicator: This contains a selected tab indicator to show that the tab is current.
+        - Optional metadata: You might display 1 item of metadata, such as a [Badge](/components/badge) to bring data forward from the content in that section or a [Tag](/components/tag) to add supplementary information, such as "New".
+    - Tab content: The content that lives "inside" a tab and is revealed when that tab item is selected.
+    - Separator: To separate tabs from content, use whitespace or the edge of another component, such as Title Blocks or Cards, not a [Divider](/components/divider).
```
Comment (didoesdigital) [inline site/docs/components/tabs.mdx:43]: Reviewing various examples for our needs, it seems more likely that we always want to use a solid line such as a divider NOT just whitespace.

File: site/docs/components/tabs.mdx [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+- Styling:
+    - Align navigation tabs and section tabs left (in left-to-right languages) and never center within a page or section.
+    - For small screens, tab items must be presented without loss of information or functionality, and without requiring scrolling in two dimensions for vertical scrolling content at a width equivalent to 320 CSS pixels and horizontal scrolling content at a height equivalent to 256 CSS pixels, as per [W3C: Understanding Success Criterion 1.4.10: Reflow](https://www.w3.org/WAI/WCAG21/Understanding/reflow.html).
```
Comment (didoesdigital) [inline site/docs/components/tabs.mdx:46]: We need a stronger stance on tabs for small screens or touch devices.

File: site/docs/components/tabs.mdx [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+    - Limit tabs to a maximum of 8 tab items and aim for 3–6. Avoid only 2 tabs.
```
Comment (didoesdigital) [inline site/docs/components/tabs.mdx:47]: More likely: "aim for 3-6", a maximum of 6, and for 2 tabs, you might consider if tabs is actually the right choice here.

File: site/docs/components/tabs.mdx [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+    - Avoid using icons in tab items.
```
Comment (didoesdigital) [inline site/docs/components/tabs.mdx:48]: "Never use icons in section tabs. If we need icons for navigation tabs on small screens, we might have stacked icons above labels."

File: site/docs/components/tabs.mdx [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+    - Tab content always appears below horizontal tabs. Tab content always appears to the right of vertical tabs in left-to-right languages and to the left of vertical tabs in right-to-left languages.
```
Comment (didoesdigital) [inline site/docs/components/tabs.mdx:49]: - Tab content always appears below horizontal tabs. 
- Tab content always appears to the right of vertical tabs in left-to-right languages (and to the left of vertical tabs in right-to-left languages).

File: site/docs/components/tabs.mdx [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+- Tab content and ordering:
+    - The first tab is selected by default. The default tab is most important use case of the page
```
Comment (didoesdigital) [inline site/docs/components/tabs.mdx:51]: Missing trailing full stop

File: site/docs/components/tabs.mdx [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+    - Use smart defaults for defining tabs that accelerate users to the content they care about most. For example, you might surface the most common pathway or personalized content, such as the current user's saved comments.
```
Comment (didoesdigital) [inline site/docs/components/tabs.mdx:52]: Whether we stick with this depends on whether tabs are used for filtering and whether tab content is always mutually exclusive.

File: site/docs/components/tabs.mdx [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+    - Avoid tabs that you can "toggle off". That is, 1 tab is always selected. If you want a tab list with no selected tabs, consider instead adding a tab representing "All" the content as the first tab.
```
Comment (didoesdigital) [inline site/docs/components/tabs.mdx:53]: Let's remove "If you want a tab list with no selected tabs, consider instead adding a tab representing "All" the content as the first tab." — that is confusing.

File: site/docs/components/tabs.mdx [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+- Navigation:
+    - A user can navigate between tabs to reveal that tab's content by clicking or tapping on the tab item. We don't support swiping between sections using the tab content area (for example, using edge swipes on mobile devices), especially when tabs have content of different length.
```
Comment (didoesdigital) [inline site/docs/components/tabs.mdx:55]: Break up the sentence at ", ". Rewrite " especially when tabs have content of different length." If we ever supported swiping between tabs, we would only support swiping the tab list itself, not the content area below tabs.

File: site/docs/components/tabs.mdx (hunk 1 above)
Comment (ActuallyACat) [reply site/docs/components/tabs.mdx:43]: Oh interesting, I thought we were trending in the other direction.

File: site/docs/components/tabs.mdx [hunk 10]
```diff
@@ -0,0 +1,115 @@
...
+- Labels:
+    - Don't truncate long tab text. Use short tab label text, and if it absolutely must be longer, consider wrapping the text instead of truncating.
+- Fixed or scrolling:
+    - Use fixed tabs of equal width for a small number of tab items (e.g. 3) that fit inside 320px containers or viewports.
+    - Use scrolling tabs of flexible width for all other tabs. That is, you can scroll them horizontally to see more tabs offscreen.
+- Overflow:
+    - Consider providing an action next to the tabs to access additional tabs that do not fit on screen. For example, you might use an adjacent menu list.
```
Comment (didoesdigital) [inline site/docs/components/tabs.mdx:62]: TODO: add this example to UI Kit

File: site/docs/components/tabs.mdx [hunk 11]
```diff
@@ -0,0 +1,115 @@
...
+- Empty state:
+    - If a tab contains no content or has an error, consider disabling or hiding it according to the [Interaction states guidelines for disabled elements](/guidelines/interaction-states#disabled-state).
```
Comment (didoesdigital) [inline site/docs/components/tabs.mdx:64]: TODO: rewrite this to not hint at 'hiding' as an option for tabs.
TODO: add a section to interaction guidelines on when you might keep a tab enabled with a badge of 0 and show an empty state component within the tab content area.

File: site/docs/components/tabs.mdx [hunk 12]
```diff
@@ -0,0 +1,115 @@
...
+- Right-To-Left (RTL):
+    - Reverse tab order for right-to-left languages.
+- Hierarchy:
+    - You might use horizontal and vertical section tabs to distinguish multiple levels of hierarchy. That is, vertical tabs might navigate between larger page sections, while horizontal tabs might navigate between data in a specific section.
```
Comment (didoesdigital) [inline site/docs/components/tabs.mdx:68]: This section needs re-thinking. The key thing we need is to establish the hierarchy in a way that designers and engineers can apply consistently that matches user mental models and expectations. Potentially, tabs is not the place to document this.

File: site/docs/components/tabs.mdx [hunk 13]
```diff
@@ -0,0 +1,115 @@
...
+<Do>
+
+- Use tabs when you have too much content to show at once. Use them to reduce cognitive load and allow users to progressively disclose the content they're interested in.
+- Use tabs for pre-defined chunks of content that users can navigate between.
+
+</Do>
+<Dont>
+
+- Avoid tabs when the items perform an action. Instead, use a [Button](/components/button) or [Button group](/components/button-group). For example, if the action filters content, use a filter instead.
+- Avoid tabs for allowing users to interact with data and reduce it themselves. Instead, use filters.
```
Comment (didoesdigital) [inline site/docs/components/tabs.mdx:83]: Both of these depend if we move forward with 'tabs as filters' being a thing

File: site/docs/components/tabs.mdx (hunk 1 above)
Comment (didoesdigital) [reply site/docs/components/tabs.mdx:43]: We were, though we're seeing more examples for our needs where the solid line creates a better sense of grouping, ensuring the tabs are perceived as closely related and completely separate from any nearby elements. For example, some tabs might appear on the left when there are also action buttons on the right.

File: site/docs/components/tabs.mdx [hunk 14]
```diff
@@ -0,0 +1,138 @@
...
+    - **Optional metadata**: You might display 1 item of metadata, such as a [Badge](/components/badge) to bring data forward from the content in that section or a [Tag](/components/tag) to add supplementary information, such as "New".
+- **Tab content**: The content that lives "inside" a tab and is revealed when that tab item is selected.
+- **Separator**: To separate tabs from content, use whitespace or the edge of another component, such as Title Blocks or Cards, not a [Divider](/components/divider).
+
+### Styling
+
+- Align navigation tabs and section tabs left (in left-to-right languages) and never center within a page or section.
+- For small screens, tab items must be presented without loss of information or functionality, and without requiring scrolling in two dimensions for vertical scrolling content at a width equivalent to 320 CSS pixels and horizontal scrolling content at a height equivalent to 256 CSS pixels, as per [W3C: Understanding Success Criterion 1.4.10: Reflow](https://www.w3.org/WAI/WCAG21/Understanding/reflow.html).
+- Never use icons in section tabs. If we need icons for navigation tabs on small screens, we might use stacked icons above labels.
```
Comment (ActuallyACat) [inline site/docs/components/tabs.mdx:50]: I'm curious for the reason for this, it feels a little arbitrary.

Comment (didoesdigital) [reply site/docs/components/tabs.mdx:50]: Reduces tab width, which is important for localisation and small screens, avoids overburdening a tab item that might already have a badge as optional metadata.

**Pull request merged**